test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -323,6 +426,98 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
+			createdPod2 := &corev1.Pod{}
+			ginkgo.By("checking that all pods in group are unsuspended when workload is admitted", func() {
+				admission := testing.MakeAdmission(clusterQueue.Name, "120fa2c0").
+					Assignment(corev1.ResourceCPU, "default", "1").
+					AssignmentPodCount(createdWorkload.Spec.PodSets[0].Count).
+					Obj()
+				gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+				util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+
+				gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
+					g.Expect(k8sClient.Get(ctx, pod1LookupKey, createdPod1)).To(gomega.Succeed())
+					return createdPod1.Spec.SchedulingGates
+				}, util.Timeout, util.Interval).ShouldNot(
+					gomega.ContainElement(corev1.PodSchedulingGate{Name: "kueue.x-k8s.io/admission"}),
+				)
+				gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
+					g.Expect(k8sClient.Get(ctx, pod2LookupKey, createdPod2)).To(gomega.Succeed())
+					return createdPod2.Spec.SchedulingGates
+				}, util.Timeout, util.Interval).ShouldNot(
+					gomega.ContainElement(corev1.PodSchedulingGate{Name: "kueue.x-k8s.io/admission"}),
+				)
+
+				gomega.Expect(len(createdPod1.Spec.NodeSelector)).Should(gomega.Equal(1))
+				gomega.Expect(createdPod1.Spec.NodeSelector["kubernetes.io/arch"]).Should(gomega.Equal("arm64"))
+				gomega.Expect(len(createdPod2.Spec.NodeSelector)).Should(gomega.Equal(1))
+				gomega.Expect(createdPod2.Spec.NodeSelector["kubernetes.io/arch"]).Should(gomega.Equal("arm64"))
+
+				gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
+				gomega.Expect(createdWorkload.Status.Conditions).Should(gomega.BeComparableTo(
+					[]metav1.Condition{
+						{Type: kueue.WorkloadQuotaReserved, Status: metav1.ConditionTrue},
+						{Type: kueue.WorkloadAdmitted, Status: metav1.ConditionTrue},
+					},
+					wlConditionCmpOpts...,
+				))
+			})
+
+			ginkgo.By("set the pods as running", func() {
+				util.SetPodsPhase(ctx, k8sClient, corev1.PodRunning, pod1, pod2)
+			})
+
+			ginkgo.By("checking that pod group is stopped when workload is evicted", func() {
+				gomega.Expect(
+					workload.UpdateStatus(ctx, k8sClient, createdWorkload, kueue.WorkloadEvicted, metav1.ConditionTrue,
+						kueue.WorkloadEvictedByPreemption, "By test", "evict"),
+				).Should(gomega.Succeed())
+				//	util.FinishEvictionForWorkloads(ctx, k8sClient, createdWorkload)
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, pod1LookupKey, createdPod1)).To(testing.BeNotFoundError())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed(), "Expected pod to be deleted")
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, pod2LookupKey, createdPod2)).To(testing.BeNotFoundError())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed(), "Expected pod to be deleted")
+			})
```

Comment:
We can add more steps to this test:

1. The workload is readmitted
2. There are new pods in the group
3. The pods succeed.

Then this would be a full test for preemption and recovery.

Reply:
done